dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SettingKey.java

```diff
@@ -245,7 +245,11 @@ public enum SettingKey {
       "keyAnalyticsCacheTtlMode", AnalyticsCacheTtlMode.FIXED, AnalyticsCacheTtlMode.class),
 
   /** Max trackedentityinstance records that can be retrieved from database. */
-  TRACKED_ENTITY_MAX_LIMIT("KeyTrackedEntityInstanceMaxLimit", 50000, Integer.class);
+  @Deprecated(forRemoval = true, since = "2.41")
+  TRACKED_ENTITY_MAX_LIMIT("KeyTrackedEntityInstanceMaxLimit", 50000, Integer.class),
+
+  /** Max tracked entity records that can be retrieved from database. */
+  TE_MAX_LIMIT("KeyTrackedEntityMaxLimit", 50000, Integer.class);
```

Comment:
I think you should be able to call this one `TRACKED_ENTITY_MAX_LIMIT` and the old one maybe `DEPRECATED_TRACKED_ENTITY_MAX_LIMIT`. I am pretty sure we recently renamed the constant to `TRACKED_ENTITY_MAX_LIMIT` removing `INSTANCE`.

Reply:
then I would suggest we keep it simple.
- change the settingkey from `KeyTrackedEntityInstanceMaxLimit` to `KeyTrackedEntityMaxLimit`
- Put a flyway script to to rename this key

Reply:
Yes, I am only talking about the variable names not the keys! The keys are exposed in our API and these need to be transitioned as @enricocolasante says. Whats internal can be changed so that its readable to us.